api/client/src/main/java/org/projectnessie/client/rest/ResponseCheckFilter.java

```diff
@@ -84,52 +86,74 @@ public static void checkResponse(ResponseContext con) throws Exception {
-      exception.addSuppressed(error.getClientProcessingException()); // for trouble-shooting
-    }
-
     throw exception;
   }
 
   private static NessieError decodeErrorObject(
-      Status status, InputStream inputStream, ObjectReader reader) {
-    NessieError error;
+      Status status, ResponseContext requestCtx, InputStream inputStream, ObjectReader reader) {
+    ImmutableNessieError.Builder error =
+        ImmutableNessieError.builder()
+            .status(status.getCode())
+            .reason(status.getReason())
+            .message(
+                requestCtx.getRequestedMethod() + " " + requestCtx.getRequestedUri() + " failed");
+
     if (inputStream == null) {
-      error =
-          ImmutableNessieError.builder()
-              .errorCode(ErrorCode.UNKNOWN)
-              .status(status.getCode())
-              .reason(status.getReason())
-              .message("Could not parse error object in response.")
-              .clientProcessingException(
-                  new RuntimeException("Could not parse error object in response."))
-              .build();
-    } else {
-      try {
-        JsonNode errorData = reader.readTree(inputStream);
-        try {
-          error = reader.treeToValue(errorData, NessieError.class);
-        } catch (JsonProcessingException e) {
-          // If the error payload is valid JSON, but does not represent a NessieError, it is likely
-          // produced by Quarkus and contains the server-side logged error ID. Report the raw JSON
-          // text to the caller for trouble-shooting.
-          error =
-              ImmutableNessieError.builder()
-                  .message(errorData.toString())
-                  .status(status.getCode())
-                  .reason(status.getReason())
-                  .clientProcessingException(e)
-                  .build();
-        }
-      } catch (IOException e) {
-        error =
-            ImmutableNessieError.builder()
-                .status(status.getCode())
-                .reason(status.getReason())
-                .clientProcessingException(e)
-                .build();
+      // jdk8.UrlConnectionRequest seems to have a null InputStream for empty responses
+      return error
+          .clientProcessingException(
+              new IOException("HTTP response was empty (inputStream was null)"))
+          .build();
+    }
+
+    final String inputString;
+    try {
+      inputString = readToString(inputStream);
+    } catch (IOException e) {
+      IOException readException = new IOException("Unable to read response body");
+      readException.addSuppressed(e);
+      return error.clientProcessingException(readException).build();
+    }
+
+    final JsonNode errorData;
+    try {
+      errorData = reader.readTree(inputString);
+    } catch (IOException e) {
+      IOException parseException =
+          new IOException("Unable to parse response body as JSON:\n" + inputString);
```

Comment:
Providing the entire input string in the error message is dangerous imho. I've seen sometimes Quarkus reply with a giant HTML. At the very least you should truncate to the first chars.

Reply:
I agree. Maybe we can have a client-side flag to control the reporting of the full error payload.

In most cases it is not required. When networking is configured correctly, error responses are proper Nessie JSON objects.

Reply:
Why not check the `Content-Type` header?